src/lib/global-context.ts

```diff
@@ -96,5 +98,9 @@ function inCompile() {
   return !!mainContext?.inCompile;
 }
 function inCheckedComputation() {
-  return !!mainContext?.inCompile || !!mainContext?.inProver;
+  return (
+    !!mainContext?.inCompile ||
+    !!mainContext?.inProver ||
+    !!mainContext?.inCheckedComputaton
+  );
```

Comment:
What's the difference between these cases now?

Given this function is named `inCheckedComputation` and we check for `inCheckedComputaton`, could we refactor `inCompile` and `inProver` to use `inCheckedComputation` instead?

Reply:
So, I added the additional inCheckedComputation because I don't yet have a good way of knowing whether we're in prove or conpile, in the case of plain Circuits. And so far, we also don't need that distinction there, we only need to be able to know if it's one of them.
In SmartContracts however, this distinction is important in some places, so we need to keep the more detailed information provided by inProver / inCompile

Reply:
For example, when doing this.state.get(), behaviour is different for inProver vs inCompile

inCheckedComputation just means "one of the two". This is the info we need for Poseidon. We also have an OCaml way of getting this info, but that was buggy in the past, so I'm trying to avoid it now

For Poseidon, we just need to know whether we

Reply:
See https://github.com/o1-labs/snarkyjs/issues/65
I'm really scared of this breaking things and causing some of our known bugs, we should dig into it soon